app/controllers/comment_controller.rb

```diff
@@ -1,3 +1,16 @@
+class CommentError < ArgumentError
+end
+
+def create_comment(node, user, body)
```

Comment:
Okay, If @jywarren approves the function, I would move it to the corresponding helper class.